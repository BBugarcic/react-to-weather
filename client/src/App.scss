// variables
$primary-color: red;
$column-width: 50%;
$column-height: 600px;

// app style
html,
body {
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  overflow: hidden;
}

#root {
  height: 100%;
}

.dashboard {
  height: 100%;
  display: flex;
  flex-direction: row-reverse;
  text-align: left;
  justify-content: center;
  align-items: center;
}

.infoPanel {
  height: $column-height;
  width: $column-width;
  display: flex;
  justify-content: flex-end;
  padding-right: 50px;
  .info {
    width: 350px;
    background-color: lightgreen;
  }
}

.navPanels {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: $column-width;
  height: $column-height;
  text-transform: capitalize;
  .selectedCity {
    padding: 20px;
  }
}

.navPanel {
  margin: 0;
  padding: 10px 20px;
  border-left: solid 1px #cccccc73;
  color: #cccccc73;
  font-weight: 300;
  cursor: pointer;
  &.selected {
    color: rgb(123, 123, 123);
    border-left: solid 1px rgb(123, 123, 123);
  }
}

.wi {
  font-size: 48px;
  color: red;
}
