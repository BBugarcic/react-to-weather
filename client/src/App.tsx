import React, { Component } from 'react'
import BarLoader from 'react-spinners/BarLoader'
import 'weather-icons/css/weather-icons.css'
import './App.scss'
import WeatherDashboard from './components/WeatherDashboard'
import { spinnerStyle } from './styles/cssInJsConstants'
import { getWeatherInfo } from './utilities/apiCalls'

interface AppState {
  isLoaded: boolean
  data: any
  error: any
}

type Keys =
  | 'apparentTemperature'
  | 'cloudCover'
  | 'dewPoint'
  | 'humidity'
  | 'nearestStormDistance'
  | 'ozone'
  | 'precipIntensity'
  | 'precipIntensityError'
  | 'precipProbability'
  | 'pressure'
  | 'temperature'
  | 'time'
  | 'uvIndex'
  | 'visibility'
  | 'windGust'
  | 'windSpeed'
  | 'icon'
  | 'precipType'
  | 'summary'

export interface CurrentWeather {
  currently: {
    [K in Keys]: number | string
  }
  latitude: number
  longitude: number
  offset: number
  timezone: string
}

class App extends Component<{}, AppState> {
  constructor(props: {}) {
    super(props)
    this.state = {
      error: null,
      isLoaded: false,
      data: null,
    }
  }

  async componentDidMount() {
    const data: CurrentWeather = await getWeatherInfo()
    this.setState({
      data: data,
      isLoaded: !!data,
    })
  }

  render() {
    const { isLoaded, data } = this.state
    if (isLoaded) {
      return <WeatherDashboard />
    } else {
      return <BarLoader css={spinnerStyle} />
    }
  }
}

export default App
