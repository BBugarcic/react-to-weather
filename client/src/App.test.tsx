import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { mount } from 'enzyme'
import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import App from './App'

const response = {
  data: {
    currently: {
      apparentTemperature: 13,
      cloudCover: 13,
      dewPoint: 13,
      humidity: 13,
      nearestStormDistance: 13,
      ozone: 13,
      precipIntensity: 13,
      precipIntensityError: 13,
      precipProbability: 13,
      pressure: 13,
      temperature: 13,
      time: 1311111111,
      uvIndex: 13,
      visibility: 13,
      windGust: 13,
      windSpeed: 13,
      icon: 'cloudy',
      precipType: 13,
      summary: 'test',
    },
    latitude: 13,
    longitude: 13,
    offset: 1,
    timezone: 'Europa/London',
    city: 'london',
  },
}

describe('<App />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div')
    act(() => {
      ReactDOM.render(<App />, div)
    })
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders error message when api returns error', done => {
    const mock = new MockAdapter(axios)
    mock.onGet('/currentWeather').networkError()
    act(() => {
      const component = mount(<App />)
      setImmediate(() => {
        component.update()
        done()
      })
      expect(component).toMatchSnapshot()
    })
  })

  it('renders <WeatherDashboard /> when api response is success', done => {
    const mock = new MockAdapter(axios)
    mock.onGet('/currentWeather').reply(200, response.data)
    act(() => {
      const component = mount(<App />)
      setImmediate(() => {
        component.update()
        done()
      })
      expect(component).toMatchSnapshot()
    })
  })
})
