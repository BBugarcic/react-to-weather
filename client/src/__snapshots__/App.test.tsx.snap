// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <WeatherDashboard /> when api response is success 1`] = `ReactWrapper {}`;

exports[`<App /> renders error message when api returns error 1`] = `ReactWrapper {}`;
